### PR TITLE
INTYGFV-14815: Adjustments to the validation implementation to cater for new features introduced with db.

### PR DESCRIPTION
**Pull request by mww-sopra-steria**
This PR is not completed yet and there will be some more changes.

### CODE REVIEW

File: packages/webcert/src/store/certificate/certificateMiddleware.test.ts
Comment (SamuelTilly) [inline packages/webcert/src/store/certificate/certificateMiddleware.test.ts:727]: Looks like this is not used so should probably be removed, but if it was used it could be solved with `testStore.dispatch(updateUser({ ...getUser(), signingMethod: SigningMethod.DSS }))` without having to disable type checking.

File: packages/webcert/src/store/certificate/certificateMiddleware.test.ts
Comment (SamuelTilly) [inline packages/webcert/src/store/certificate/certificateMiddleware.test.ts:808]: Could be shortened using the fakeCertificate functions

```typescript
return {
    metadata: fakeCertificateMetaData(),
    data: fakeRadioBooleanElement({
      id: '0',
      value: { selected },
      validation: [
        {
          questionId: '0',
          type: CertificateDataValidationType.SHOW_VALIDATION,
          expression: '$0',
        },
      ],
    }),
    links: [],
  }```

File: packages/webcert/src/store/certificate/certificateMiddleware.test.ts
Comment (SamuelTilly) [inline packages/webcert/src/store/certificate/certificateMiddleware.test.ts:840]: Could also be shorten

```typescript
return {
    metadata: fakeCertificateMetaData(),
    data: fakeDateElement({
      id: '0',
      value: { date },
      validation: [
        {
          questionId: '0',
          type: CertificateDataValidationType.SHOW_VALIDATION,
          expression: '$0',
        },
      ],
    }),
    links: [],
  }
```

File: packages/webcert/src/utils/structureCertificate.ts
Comment (SamuelTilly) [inline packages/webcert/src/utils/structureCertificate.ts:14]: Do we still need to run `sortByIndex` after the `.reduce` function?

File: packages/common/src/utils/validationUtils.ts
Comment (SamuelTilly) [inline packages/common/src/utils/validationUtils.ts:258]: Looks like this function is not being used, could be removed?

**Review by SamuelTilly — CHANGES_REQUESTED** [review]

File: packages/common/src/utils/validationUtils.ts
Comment (ChrilleSigerhed) [reply packages/common/src/utils/validationUtils.ts:258]: The function is getting called in the filter on line 258.

Comment (SamuelTilly) [reply packages/common/src/utils/validationUtils.ts:258]: Oh, I thought that was from inside itself, but I see now. 👍

**Review by ChrilleSigerhed — APPROVED** [review]
Looks good, well done. I suggest we merge this after Webcert & Common are finished merging!

**Review by SamuelTilly — APPROVED** [review]

**Pull request merged**
